### DISCUSSION TITLE
How can I convert grayscale to Jpeg2000?

### DISCUSSION THREAD

**Discussion by cargfiloy** [dlemstra/Magick.NET#1403] [General]
Hello~!
I need to convert 8bit grayscale image to Jpeg2000(.j2k or .jp2).
I want to control the image quality or rate.
How can I do that?

Below is the sample code I wrote.
Is it right?

----- Encoding ---------
var settings = new MagickReadSettings();
settings.Width = isoData.representationHeaders[i].ImageWidth;
settings.Height = isoData.representationHeaders[i].ImageHeight;
settings.Format = MagickFormat.Gray;
                    
MagickImage image = new MagickImage(isoData.representationHeaders[i].Image, settings);
var memStream = new MemoryStream();

image.Format = MagickFormat.J2k;
image.Settings.Compression = CompressionMethod.JPEG2000;
image.Quality = 50;                          // Quality is compression rate?
image.Write(memStream);

var data = memStream.GetBuffer();  // data has J2k?

-----Decoding ---------
var settings = new MagickReadSettings();
settings.Width = isoData.representationHeaders[i].ImageWidth;
settings.Height = isoData.representationHeaders[i].ImageHeight;
settings.Format = MagickFormat.J2k;

MagickImage image = new MagickImage(isoData.representationHeaders[i].Image, settings);
var memStream = new MemoryStream();

image.Format = MagickFormat.Gray;
image.Quality = 50;
image.Write(memStream);

var data = memStream.GetBuffer();  // data has decompress?

**Comment by dlemstra** [comment]
I would advise you to use the `Jp2WriteDefines`. This allows you to set more options.

**Reply by cargfiloy** [reply]
@dlemstra 
Thank you. This is what i wanted.
